src/CMakeLists.txt

```
@@ -6,36 +6,51 @@ add_library(NeoFOAM SHARED)
 
 include(GNUInstallDirs)
 
-target_sources(
-  NeoFOAM
-  PRIVATE "core/mpi/halfDuplexCommBuffer.cpp"
-          "core/primitives/vector.cpp"
-          "core/time.cpp"
-          "core/dictionary.cpp"
-          "core/demangle.cpp"
-          "core/tokenList.cpp"
-          "core/kokkos.cpp"
-          "executor/CPUExecutor.cpp"
-          "executor/GPUExecutor.cpp"
-          "executor/serialExecutor.cpp"
-          "mesh/unstructured/boundaryMesh.cpp"
-          "mesh/unstructured/communicator.cpp"
-          "mesh/unstructured/unstructuredMesh.cpp"
-          "finiteVolume/cellCentred/stencil/stencilDataBase.cpp"
-          "finiteVolume/cellCentred/stencil/stencilDataBase.cpp"
-          "finiteVolume/cellCentred/stencil/geometryScheme.cpp"
-          "finiteVolume/cellCentred/stencil/basicGeometryScheme.cpp"
-          "finiteVolume/cellCentred/boundary/boundary.cpp"
-          "finiteVolume/cellCentred/operators/gaussGreenGrad.cpp"
-          "finiteVolume/cellCentred/operators/gaussGreenDiv.cpp"
-          "finiteVolume/cellCentred/interpolation/linear.cpp"
-          "finiteVolume/cellCentred/interpolation/upwind.cpp")
+set(NeoFOAM_SRCS
+    "core/mpi/halfDuplexCommBuffer.cpp"
+    "core/primitives/vector.cpp"
+    "core/time.cpp"
+    "core/dictionary.cpp"
+    "core/demangle.cpp"
+    "core/tokenList.cpp"
+    "core/kokkos.cpp"
+    "executor/CPUExecutor.cpp"
+    "executor/GPUExecutor.cpp"
+    "executor/serialExecutor.cpp"
+    "mesh/unstructured/boundaryMesh.cpp"
+    "mesh/unstructured/communicator.cpp"
+    "mesh/unstructured/unstructuredMesh.cpp"
+    "finiteVolume/cellCentred/stencil/stencilDataBase.cpp"
+    "finiteVolume/cellCentred/stencil/stencilDataBase.cpp"
+    "finiteVolume/cellCentred/stencil/geometryScheme.cpp"
+    "finiteVolume/cellCentred/stencil/basicGeometryScheme.cpp"
+    "finiteVolume/cellCentred/boundary/boundary.cpp"
+    "finiteVolume/cellCentred/operators/gaussGreenGrad.cpp"
+    "finiteVolume/cellCentred/operators/gaussGreenDiv.cpp"
+    "finiteVolume/cellCentred/interpolation/linear.cpp"
+    "finiteVolume/cellCentred/interpolation/upwind.cpp")
+
+target_sources(NeoFOAM PRIVATE ${NeoFOAM_SRCS})
+if(Kokkos_ENABLE_CUDA)
+  set_source_files_properties(${NeoFOAM_SRCS} PROPERTIES LANGUAGE CUDA)
+endif()
 
 include(${CMAKE_SOURCE_DIR}/cmake/Sanitizer.cmake)
 enable_sanitizers(
   NeoFOAM NEOFOAM_ENABLE_SANITIZE_ADDRESS NEOFOAM_ENABLE_SANITIZE_LEAK NEOFOAM_ENABLE_SANITIZE_UB
   NEOFOAM_ENABLE_SANITIZE_THREAD NEOFOAM_ENABLE_SANITIZE_MEMORY)
 
 target_link_libraries(NeoFOAM PRIVATE neofoam_warnings neofoam_options)
-target_link_libraries(NeoFOAM PUBLIC NeoFOAM_public_api MPI::MPI_CXX Kokkos::kokkos
+target_link_libraries(NeoFOAM PUBLIC NeoFOAM_public_api MPI::MPI_C MPI::MPI_CXX Kokkos::kokkos
                                      cpptrace::cpptrace)
+
+add_library(NeoFOAM-STATIC)
```

Would this work? the rest of the CMake variables also use underscores.
```suggestion
add_library(NeoFOAM_STATIC)
```

---

yes, works